packages/components/src/Notification/Notification.component.js

```diff
@@ -172,6 +172,14 @@ class Registry {
 			this.timerRegistry[notification.id].cancel();
 		}
 	}
+
+	clean(notification) {
+		if (this.isRegistered(notification)) {
+			this.cancel(notification);
+			// Clean registry to avoid memory leak
+			delete this.timerRegistry[notification.id];
```

Comment:
Shouldn't it be removed on cancel too ?
From what I saw in the code, on close, we call cancel, but the timer is not removed in this case.

So what about : 
- cancel: cancel the timer, and remove it from registry
- On close: cancel
- On register, the timer calls cancel instead of this clean function. If we move the delete in the cancel function, this clean function is not relevant anymore